cannot find OS family for guest ID "" when deploying from content library

### Terraform Version
```
❯ terraform version
Terraform v0.12.28
+ provider.random v2.2.1
+ provider.vsphere v1.21.0
```
### vSphere Provider Version
+ provider.vsphere v1.21.0


### Affected Resource(s)

<!--- Please list the resources as a list, for example:
- `vsphere_virtual_machine`
- `vsphere_distributed_port_group`

If this issue appears to affect multiple resources, it may be an issue with
Terraform's core, so please mention this.
--->

vsphere_virtual_machine

### Terraform Configuration Files

<!--- Information about code formatting: https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code --->

```hcl
data "vsphere_content_library" "library" {
  name = var.vsphere_content_library
}

data "vsphere_content_library_item" "image" {
  name       = var.vm_ovf_template
  library_id = data.vsphere_content_library.library.id
}

resource "vsphere_virtual_machine" "master_vm" {
  name             = join("-", [var.vm_prefix, "master"])
  resource_pool_id = data.vsphere_resource_pool.pool.id
  datastore_id     = data.vsphere_datastore.datastore.id

  num_cpus = var.vm_cpus
  memory   = var.vm_memory
  guest_id = "otherLinux64Guest"

  network_interface {
    network_id   = data.vsphere_network.network.id
  }

  disk {
    label            = "disk0"
    size             = var.vm_disk_size
    thin_provisioned = false
  }

  clone {
    template_uuid = data.vsphere_content_library_item.image.id
    customize {
      linux_options {
        host_name = join("-", [var.vm_prefix, "master"])
        domain = var.vm_domain
      }
      network_interface {}
    }
  }
}
```

### Debug Output

Before accepting the values

```
  # vsphere_virtual_machine.master_vm will be created
  + resource "vsphere_virtual_machine" "master_vm" {
      + boot_retry_delay                        = 10000
      + change_version                          = (known after apply)
      + cpu_limit                               = -1
      + cpu_share_count                         = (known after apply)
      + cpu_share_level                         = "normal"
      + datastore_id                            = "datastore-48"
      + default_ip_address                      = (known after apply)
      + ept_rvi_mode                            = "automatic"
      + firmware                                = "bios"
      + force_power_off                         = true
      + guest_id                                = "otherLinux64Guest"
```

When deploying

```
2020/07/01 11:51:20 [DEBUG] vsphere_virtual_machine.master_vm: apply errored, but we're indicating that via the Error pointer rather than returning it: cannot find OS family for guest ID "": could not find guest ID ""
2020/07/01 11:51:20 [ERROR] <root>: eval: *terraform.EvalApplyPost, err: cannot find OS family for guest ID "": could not find guest ID ""
2020/07/01 11:51:20 [ERROR] <root>: eval: *terraform.EvalSequence, err: cannot find OS family for guest ID "": could not find guest ID ""
```
This causes

```
Error: cannot find OS family for guest ID "": could not find guest ID ""

  on compute.tf line 23, in resource "vsphere_virtual_machine" "master_vm":
  23: resource "vsphere_virtual_machine" "master_vm" {
```

### Panic Output

<!--- If Terraform produced a panic, please provide a link to a GitHub Gist containing
the output of the `crash.log`.--->

### Expected Behavior

It should have used the guest_id I provided in the template

### Actual Behavior

It was set to null. I believe the cause of the issue is here: 

https://github.com/hashicorp/terraform-provider-vsphere/pull/1014

As the reviewer mentioned, it should first check if the value exists before nulling that out.  cc @bill-rich @Satak

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

<!--- Are there anything atypical about your infrastructure that we should know? --->

### References

https://github.com/hashicorp/terraform-provider-vsphere/pull/1014

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

--- a/vsphere/resource_vsphere_virtual_machine.go
+++ b/vsphere/resource_vsphere_virtual_machine.go
@@ -1003,6 +1003,10 @@ func resourceVSphereVirtualMachineCustomizeDiff(d *schema.ResourceDiff, meta int
 			}
 			fallthrough
 		default:
+			// If guest_id is not set and the source is not a Content Library item, set it to the default.
+			if d.Get("guest_id") == "" {
+				d.SetNew("guest_id", "other-64")
+			}
 			// For most cases (all non-imported workflows), any changed attribute in
 			// the clone configuration namespace is a ForceNew. Flag those now.
 			for _, k := range d.GetChangedKeysPrefix("clone.0") {
@@ -1662,6 +1666,14 @@ func resourceVSphereVirtualMachineCreateClone(d *schema.ResourceData, meta inter
 			fmt.Errorf("error reconfiguring virtual machine: %s", err),
 		)
 	}
+
+	vmprops, err := virtualmachine.Properties(vm)
+	if err != nil {
+		return nil, err
+	}
+
+	// This should only change if deploying from a Content Library item.
+	d.Set("guest_id", vmprops.Config.GuestId)
 
 	// Upgrade the VM's hardware version if needed.
 	err = virtualmachine.SetHardwareVersion(vm, d.Get("hardware_version").(int))

--- a/vsphere/virtual_machine_config_structure.go
+++ b/vsphere/virtual_machine_config_structure.go
@@ -252,7 +252,7 @@ func schemaVirtualMachineConfigSpec() map[string]*schema.Schema {
 		"guest_id": {
 			Type:        schema.TypeString,
 			Optional:    true,
-			Default:     "other-64",
+			Computed:    true,
 			Description: "The guest ID for the operating system.",
 			DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
 				if len(d.Get("ovf_deploy").([]interface{})) > 0 {

